go/proxyd/integration_tests/batch_timeout_test.go

```diff
@@ -29,7 +29,7 @@ func TestBatchTimeout(t *testing.T) {
 	slowBackend.SetHandler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// check the config. The sleep duration should be at least double the server.timeout_seconds config to prevent flakes
 		time.Sleep(time.Second * 2)
-		SingleResponseHandler(200, goodResponse)(w, r)
```

Comment:
TODO: this test fails as we no longer short-circuit when we've reached the deadline.

Reply:
Isn't that something we want?